### PR TITLE
[RISCV] Refactor performCONCAT_VECTORSCombine. NFC

### PR DESCRIPTION
**Pull request by lukel97**
Instead of doing a forward pass for positive strides and a reverse pass for
negative strides, we can just do one pass by negating the offset if the
pointers do happen to be in reverse order.

We can extend getPtrDiff later in #68726 to handle more constant offset
sequences.

### CODE REVIEW

**Review by michaelmaitland — APPROVED** [review]
Thanks for making this change! LGTM.

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -13798,58 +13797,33 @@ static SDValue performCONCAT_VECTORSCombine(SDNode *N, SelectionDAG &DAG,
...
-        return SDValue();
-    }
-    return Stride;
+  auto getPtrDiff = [&DAG, &DL](LoadSDNode *Ld1, LoadSDNode *Ld2) {
+    SDValue P1 = Ld1->getBasePtr();
+    SDValue P2 = Ld2->getBasePtr();
+    if (P2.getOpcode() == ISD::ADD && P2.getOperand(0) == P1)
+      return P2.getOperand(1);
+    if (P1.getOpcode() == ISD::ADD && P1.getOperand(0) == P2)
+      return DAG.getNegative(P1.getOperand(1), DL,
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:13812]: Do we want to create this node if the overall match fails?

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -13867,26 +13844,25 @@ static SDValue performCONCAT_VECTORSCombine(SDNode *N, SelectionDAG &DAG,
   if (!TLI.isLegalStridedLoadStore(WideVecVT, Align))
     return SDValue();
 
+  auto [Stride, Reversed] = *BaseDiff;
```
[the remark this answers is not in the document]
Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:13847]: I am content with this approach.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**